# Hypothesis Function

![Formula](Image/Formula.png)

First, we will see the example which we had already seen in the previous blog.

### h(x) = X1\*w1 + X2\*w2 + X3\*w3 + b

In [1]:
# importing numpy
import numpy as np

In [2]:
# values of X
X = np.array([0.3,0.5,0.6])
# values of weights and bias
w = np.array([0.02,0.2,0.1,-0.3]) # here b = w0

In [3]:
# Add x0 = 1 to the value of x to multiply with the bias(b) value
X = np.insert(X,0,1)

In [4]:
# hypothesis function
def hy_function(x,w):
    return np.dot(x,w)

In [9]:
y_hat = hy_function(X,w) # y_hat is the prection value which is equal to h(x)
print(y_hat)

-0.04999999999999999


Here we found the value -0.049999... which is approximately equal to -0.05 as we had seen in the previous blog.

Now let's try another example with the same x values but randomly generated weights and bias value in between -1 to 1 (for fun)

In [6]:
# function for randomly generated weights and bias in between -1 to 1
def random_weights(x):
    return (np.random.rand(x.size)*2)-1

In [7]:
# random weight
random_wb = random_weights(X)

In [8]:
# predicted y_hat value
y_hat = hy_function(X,random_wb)
print(y_hat)

1.1646668378490117


So, the predicted value with random weight is 1.16

Thank you <p align="right"><a href="3.Activation-Function.md">Next>></a></p>